sphinx/util/inspect.py

```diff
@@ -557,6 +557,45 @@ def signature_from_str(signature: str) -> inspect.Signature:
     return inspect.Signature(params, return_annotation=return_annotation)
 
 
+def signature_from_ast(node: ast.FunctionDef) -> inspect.Signature:
+    """Return a Signature object for the given AST *node*."""
+    params = []
+    if hasattr(node.args, "posonlyargs"):  # for py38+
+        for arg in node.args.posonlyargs:  # type: ignore
+            param = Parameter(arg.arg, Parameter.POSITIONAL_ONLY,
+                              annotation=ast_unparse(arg.annotation))
+            params.append(param)
```

Comment:
Positional only arguments can have defaults too

Reply:
OMG, this comes from another bug...
https://github.com/sphinx-doc/sphinx/issues/7756